core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -177,7 +177,9 @@ public void initialize(String name, Map<String, String> unresolved) {
     this.io =
         CatalogUtil.loadFileIO(
             ioImpl != null ? ioImpl : ResolvingFileIO.class.getName(), mergedProps, conf);
-    this.reporter = new LoggingMetricsReporter();
+    String metricsReporterImpl = mergedProps.get(CatalogProperties.METRICS_REPORTER_IMPL);
```

Comment:
Would it make sense to add a test where you configure a custom MetricsReporter through this new catalog property?

Reply:
makes sense, I added a test